packages/portal/document-viewer/src/utils/addUniqueHeadingSlugs.ts

```diff
@@ -0,0 +1,27 @@
+import { toSlug } from './toSlug'
+export function addUniqueHeadingSlugs<T = any>(children: Array<T>): Array<T> {
+
+  const slugMap = new Map<string, number>()
+  // loop through the children and their children recursively
```

Comment:
nit: we went with not needing to loop recursively
```suggestion
  // loop through the children
```